File: packages/zynth-webserver/ios/ZynthWebServer.swift
//
//  ZynthWebServer.swift
//  ZynthWebServer
//
//  Public interface for ZynthWebServer module
//

import Foundation
import ZynthKit

@objc(ZynthWebServer)
public class ZynthWebServer: NSObject {
  private static var moduleInstance: ZynthWebServerModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      print("[ZynthWebServer] Module already initialized")
      return
    }

    let module = ZynthWebServerModule()
    runtime.installModules([module])
    moduleInstance = module
    print("[ZynthWebServer] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}

File: packages/zynth-webserver/android/ZynthWebServer/src/main/java/dev/zynth/webserver/ZynthWebServer.kt
package dev.zynth.webserver

import android.app.Activity
import android.util.Log
import com.zynth.kit.runtime.ZynthRuntime

/**
 * Public interface for ZynthWebServer module.
 */
object ZynthWebServer {
    private var moduleInstance: ZynthWebServerModule? = null

    @JvmStatic
    fun initialize(activity: Activity, runtime: ZynthRuntime) {
        if (moduleInstance != null) {
            Log.w("ZynthWebServer", "Module already initialized")
            return
        }

        moduleInstance = ZynthWebServerModule(activity)
        runtime.installModules(listOf(moduleInstance!!))
        Log.d("ZynthWebServer", "Module initialized")
    }

    @JvmStatic
    fun cleanup() {
        moduleInstance?.shutdown()
        moduleInstance = null
    }
}
